src/cl_main.c

```diff
@@ -808,13 +808,33 @@ void CL_Connect_f (void)
 	qbool proxy;
 	char *connect_addr = NULL;
 	char *server_buf = NULL;
+	portpingprobe_status_t status;
 
 	if (Cmd_Argc() != 2) 
 	{
 		Com_Printf ("Usage: %s <server>\n", Cmd_Argv(0));
 		return;
 	}
 
+	// It's not possible to change the source port when connected to a
+	// server, so we'll inform the user and return early
+	if (UsePortPingProbe() && cls.state != ca_disconnected)
+	{
+		Com_Printf("With cl_useportpingprobe enabled, you must disconnect before switching servers\n");
```

Comment:
Could we disconnect the client automatically before connecting to another server when ping probe is enabled?

I suppose this will act different in case the server address we `connect` to can't be found thoughg.

Reply:
Yes, this can be changed. I just thought it would be less intrusive for the user to implement it this way

Reply:
I received another request for this change, so I modified the code to automatically disconnect if you are already connected.